.github/workflows/runtests.yml

```
@@ -69,28 +79,78 @@ jobs:
+
+      - name: save free-threaded test data cache
+        if: matrix.free_thread && steps.restore-free-thread-test-data.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ env.DATA_CACHE_PATH }}
+          key: ${{ env.DATA_CACHE_KEY }}
+
       - name: generate qmd docs tests
         id: generate_qmd_tests
-        shell: bash -el {0}
+        shell: ${{ matrix.test_shell }}
```

actionlint (1.7.12)

[error] context "matrix" is not allowed here. no context is available here. see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)